src/config.c

```diff
@@ -248,26 +283,34 @@ dictType optionToLineDictType = {
 };
 
 dictType optionSetDictType = {
-    dictSdsCaseHash,            /* hash function */
+    distCStrCaseHash,           /* hash function */
     NULL,                       /* key dup */
     NULL,                       /* val dup */
-    dictSdsKeyCaseCompare,      /* key compare */
+    dictCStrKeyCaseCompare,     /* key compare */
     dictSdsDestructor,          /* key destructor */
     NULL,                       /* val destructor */
     NULL                        /* allow to expand */
 };
 
-dictType sdsKeyValueDictType = {
-    dictSdsCaseHash,            /* hash function */
+dictType configSetDictType = {
+    distCStrCaseHash,           /* hash function */
     NULL,                       /* key dup */
     NULL,                       /* val dup */
-    dictSdsKeyCaseCompare,      /* key compare */
+    dictCStrKeyCaseCompare,     /* key compare */
     dictSdsDestructor,          /* key destructor */
```

Comment:
Here I'm not sure we need the key to be a duped sds. It can be simply a pointer to the cstr which doesn't need to be de-allocated.

Reply:
currently we are adding sdsnew(config->name), config->name is const char *, do you suggest allocating a char * and adding that to the dictionary as the key?